File: docs/WidgetMill/flutter/sdui_flutter/lib/src/builders/wrap.dart
import 'package:flutter/widgets.dart';
import '../model/sdui_node.dart';
import '../parsing/enums.dart';
import '../parsing/parsers.dart';
import '../renderer.dart';

/// Wrap — flui os filhos em múltiplas linhas/colunas.
Widget buildWrap(BuildContext context, SduiNode node, SduiRenderer r) {
  final p = node.props;
  return Wrap(
    direction: axisFrom(p['direction']),
    alignment: wrapAlignmentFrom(p['alignment']),
    runAlignment: wrapAlignmentFrom(p['runAlignment']),
    crossAxisAlignment: wrapCrossFrom(p['crossAxisAlignment']),
    spacing: parseDouble(p['spacing']) ?? 0,
    runSpacing: parseDouble(p['runSpacing']) ?? 0,
    children: r.renderAll(context, node.children),
  );
}

File: docs/WidgetMill/packages/spec/src/nodes/wrap.ts
import { z } from "zod";
import { AXIS, RUN_ALIGNMENT, WRAP_ALIGNMENT, WRAP_CROSS } from "../enums";
import { NodeRef } from "../node-ref";

/** Wrap — flui os filhos em múltiplas linhas/colunas (catálogo Flutter). */
export const WrapNode = z.object({
  type: z.literal("wrap"),
  props: z
    .object({
      direction: z.enum(AXIS).optional(),
      alignment: z.enum(WRAP_ALIGNMENT).optional(),
      runAlignment: z.enum(RUN_ALIGNMENT).optional(),
      crossAxisAlignment: z.enum(WRAP_CROSS).optional(),
      spacing: z.number().optional(),
      runSpacing: z.number().optional(),
    })
    .default({}),
  children: z.array(NodeRef).default([]),
});
